components/query.py

```diff
@@ -139,7 +158,13 @@ def get_filenames(df, subspecies, view, sex, hybrid, num_images):
     df_sub = df_sub.loc[df_sub.View.isin(view)]
     df_sub = df_sub.loc[df_sub.Sex.isin(sex)]
     df_sub = df_sub.loc[df_sub.hybrid_stat.isin(hybrid)]
+
+    num_entries = len(df_sub)
+    # Filter out any entries that have missing filenames or URLs:
+    df_sub = df_sub.loc[df_sub.Image_filename != 'unknown']
+    df_sub = df_sub.loc[df_sub.file_url != 'unknown']
```

Comment:
I'm wondering whether these steps wouldn't best happen as the first (before any filtering steps) rather than later on. For the smallish datasets were dealing with here, it probably makes absolutely no difference, but it's not a bad idea to keep the possibility of bigger ones in mind.

Reply:
I had put it after since I was thinking of returning the number of entries that exist when there's no filename or path to display the images, as this would still provide some information. If that's not information we want to maintain, then yes, for efficiency it might make sense to filter the unknowns first.